### PR TITLE
[Material] Fix a jumping animation in the beginning of the extended Navigation Rail transition

### LINKED ISSUE #65657: [Navigation Rail] Extended animation jumps on initially frame
As soon as the animation is non-zero, a const SizedBox is inserted which causing an unsmooth animation.

nav

### PR DESCRIPTION
**Pull request by clocksmith**
## Description

This change lerps the width in a SizedBox that is used as spacing in the extended nav rail, rather than using a constant value. This fixes a bug in the animation of the extended rail, which also can cause inconsistencies with the leading/trailing widgets.

Before (slowed down 15x):
bug_bad

After fix (slowed down 15x):
bug_good

Note that the change is subtle, but it is more apparent, and causes more issues, when there are animated widgets in the leading/trailing slots of the rail.

## Related Issues

closes https://github.com/flutter/flutter/issues/65657

## Tests

I added the following tests:

I added 'Extended rail transition does not jump from the beginning' to navigation_rail_test.dart

## Checklist

Before you create this PR, confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [ ] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [x] I got input from the developer relations team, specifically from: redbrogden
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

Since the animation does not jump, the width of the rail is slightly different mid-animation, so one of the width values had to be updated.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/test/material/navigation_rail_test.dart [hunk 1]
```diff
@@ -1794,6 +1794,63 @@ void main() {
     expect(rail.size.width, equals(300.0));
   });
 
+  testWidgets('Extended rail transition does not jump from the beginning', (WidgetTester tester) async {
```
Comment (perclasson) [inline packages/flutter/test/material/navigation_rail_test.dart:1797]: Nice test

**Review by perclasson — APPROVED** [review]
LGTM

File: packages/flutter/test/material/navigation_rail_test.dart (hunk 1 above)
Comment (clocksmith) [reply packages/flutter/test/material/navigation_rail_test.dart:1797]: TY!

**Review by Renzo-Olivares — APPROVED** [review]
LGTM

**Review by berrylawchingman — APPROVED** [review]

**Review by berrylawchingman — APPROVED** [review]

File: packages/flutter/test/material/navigation_rail_test.dart [hunk 2]
```diff
@@ -1794,6 +1794,63 @@ void main() {
...
+              ),
+            );
+          },
+        ),
+      ),
+    );
+
+    final RenderBox rail = tester.firstRenderObject<RenderBox>(find.byType(NavigationRail));
+
+    expect(rail.size.width, equals(72.0));
```
Comment (HansMuller) [inline packages/flutter/test/material/navigation_rail_test.dart:1838]: Might be slightly simpler (here and below) without saving a ref to the renderobject. Also the "equals" isn't needed.
```dart
final Finder rail = find.byType(NavigationRail);
expect(tester.getSize(rail).width, 72)
```

**Review by HansMuller — APPROVED** [review]
LGTM

**Review by rami-a — APPROVED** [review]

**Pull request merged**